cms/djangoapps/contentstore/views/course.py

```diff
@@ -469,6 +472,56 @@ def _get_rerun_link_for_item(course_key):
     return reverse_course_url('course_rerun_handler', course_key)
 
 
+def _ora1_deprecation_info(course_id, advanced_modules):
+    """
+    Returns information about ORA1
+        * Modules presence in the Advanced Module List.
+        * Components presence in the course outline.
+
+    Arguments:
+        course_id (CourseKey): course id
+        advanced_modules (list): advance module list
+
+    Returns:
+        Dict with following keys:
+
+        ora1_enabled (bool): True if "peergrading" and/or "combinedopenended" are in the Advanced Module List else False
+        ora1_components (list): List of ora1 components and their parent's url
+        advance_settings_url (str): URL to advance settings page
+    """
+    data = {
+        'ora1_enabled': 'peergrading' in advanced_modules or 'combinedopenended' in advanced_modules,
+        'ora1_components': [],
+        'advance_settings_url': reverse_course_url('advanced_settings_handler', course_id)
+    }
+
+    cache_key = 'ora1.components.{course}'.format(course=course_id)
+
+    ora1_components = cache.get(cache_key)  # pylint: disable=maybe-no-member
+    if not ora1_components:
+        try:
+            course_structure = CourseStructure.objects.get(course_id=course_id)
+            ordered_blocks = course_structure.ordered_blocks
+        except CourseStructure.DoesNotExist:
+            return data
+
+        store = modulestore()
+        ora1_components = []
+        for __, block in ordered_blocks.items():
+            if block['block_type'] in ['peergrading', 'combinedopenended']:
+                item = store.get_item(UsageKey.from_string(block['parent']))
+                if not getattr(item, 'is_draft', False):
```

Comment:
@ormsbee Here is the scenario
1. Create a unit with ORA1 components and publish it
2. Visit the course outline and you will see a list of ORA1 components 
3. Now create two more units with ORA1 components, don't publish them
4. Visit the course outline and you will see the same ORA1 components as you seen in `step 2` 
5. Now publish any of unit from step 3
6. Visit the course outline and you will see all(published and unpublished) the ORA1 components IF THERE IS NO CHECK OF `is_draft` 
7. The problem is behavior is different in `step 4` and `step 6` thats why we added the `is_draft` check to always show the published items